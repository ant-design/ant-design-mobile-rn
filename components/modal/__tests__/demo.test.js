import rnDemoTest from '../../../tests/shared/demoTest'
// https://github.com/facebook/jest/issues/6434#issuecomment-525576660
jest.useFakeTimers()
rnDemoTest('modal')
