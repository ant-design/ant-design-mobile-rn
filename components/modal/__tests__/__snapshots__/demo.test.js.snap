// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ./components/modal/demo/basic.tsx correctly 1`] = `
<ScrollView
  style={
    Object {
      "marginTop": 20,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
      ]
    }
  >
    <
      activeOpacity={0.4}
      disabled={false}
      onHideUnderlay={[Function]}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      onShowUnderlay={[Function]}
      pressIn={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "borderWidth": 1,
            "justifyContent": "center",
          },
          Object {
            "height": 47,
            "paddingLeft": 15,
            "paddingRight": 15,
          },
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#dddddd",
          },
          false,
          false,
          undefined,
          undefined,
        ]
      }
      underlayColor="#dddddd"
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#000000",
              },
              false,
              false,
            ]
          }
        >
          showModal
        </Text>
      </View>
    </>
    <View
      style={
        Array [
          Object {
            "height": 9,
          },
          undefined,
        ]
      }
    />
    <
      activeOpacity={0.4}
      disabled={false}
      onHideUnderlay={[Function]}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      onShowUnderlay={[Function]}
      pressIn={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "borderWidth": 1,
            "justifyContent": "center",
          },
          Object {
            "height": 47,
            "paddingLeft": 15,
            "paddingRight": 15,
          },
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#dddddd",
          },
          false,
          false,
          undefined,
          undefined,
        ]
      }
      underlayColor="#dddddd"
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#000000",
              },
              false,
              false,
            ]
          }
        >
          transparent:false
        </Text>
      </View>
    </>
    <View
      style={
        Array [
          Object {
            "height": 9,
          },
          undefined,
        ]
      }
    />
    <
      activeOpacity={0.4}
      disabled={false}
      onHideUnderlay={[Function]}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      onShowUnderlay={[Function]}
      pressIn={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "borderWidth": 1,
            "justifyContent": "center",
          },
          Object {
            "height": 47,
            "paddingLeft": 15,
            "paddingRight": 15,
          },
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#dddddd",
          },
          false,
          false,
          undefined,
          undefined,
        ]
      }
      underlayColor="#dddddd"
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#000000",
              },
              false,
              false,
            ]
          }
        >
          popup
        </Text>
      </View>
    </>
    <View
      style={
        Array [
          Object {
            "height": 9,
          },
          undefined,
        ]
      }
    />
    <
      activeOpacity={0.4}
      disabled={false}
      onHideUnderlay={[Function]}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      onShowUnderlay={[Function]}
      pressIn={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "borderWidth": 1,
            "justifyContent": "center",
          },
          Object {
            "height": 47,
            "paddingLeft": 15,
            "paddingRight": 15,
          },
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#dddddd",
          },
          false,
          false,
          undefined,
          undefined,
        ]
      }
      underlayColor="#dddddd"
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#000000",
              },
              false,
              false,
            ]
          }
        >
          Modal.alert
        </Text>
      </View>
    </>
    <View
      style={
        Array [
          Object {
            "height": 9,
          },
          undefined,
        ]
      }
    />
    <
      activeOpacity={0.4}
      disabled={false}
      onHideUnderlay={[Function]}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      onShowUnderlay={[Function]}
      pressIn={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "borderWidth": 1,
            "justifyContent": "center",
          },
          Object {
            "height": 47,
            "paddingLeft": 15,
            "paddingRight": 15,
          },
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#dddddd",
          },
          false,
          false,
          undefined,
          undefined,
        ]
      }
      underlayColor="#dddddd"
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#000000",
              },
              false,
              false,
            ]
          }
        >
          Modal.opertation
        </Text>
      </View>
    </>
    <View
      style={
        Array [
          Object {
            "height": 9,
          },
          undefined,
        ]
      }
    />
    <
      activeOpacity={0.4}
      disabled={false}
      onHideUnderlay={[Function]}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      onShowUnderlay={[Function]}
      pressIn={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "borderWidth": 1,
            "justifyContent": "center",
          },
          Object {
            "height": 47,
            "paddingLeft": 15,
            "paddingRight": 15,
          },
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#dddddd",
          },
          false,
          false,
          undefined,
          undefined,
        ]
      }
      underlayColor="#dddddd"
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#000000",
              },
              false,
              false,
            ]
          }
        >
          Modal.opertation (onBackHandler)
        </Text>
      </View>
    </>
    <View
      style={
        Array [
          Object {
            "height": 9,
          },
          undefined,
        ]
      }
    />
    <
      activeOpacity={0.4}
      disabled={false}
      onHideUnderlay={[Function]}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      onShowUnderlay={[Function]}
      pressIn={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "borderWidth": 1,
            "justifyContent": "center",
          },
          Object {
            "height": 47,
            "paddingLeft": 15,
            "paddingRight": 15,
          },
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#dddddd",
          },
          false,
          false,
          undefined,
          undefined,
        ]
      }
      underlayColor="#dddddd"
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#000000",
              },
              false,
              false,
            ]
          }
        >
          Modal.prompt (default)
        </Text>
      </View>
    </>
    <View
      style={
        Array [
          Object {
            "height": 9,
          },
          undefined,
        ]
      }
    />
    <
      activeOpacity={0.4}
      disabled={false}
      onHideUnderlay={[Function]}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      onShowUnderlay={[Function]}
      pressIn={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "borderWidth": 1,
            "justifyContent": "center",
          },
          Object {
            "height": 47,
            "paddingLeft": 15,
            "paddingRight": 15,
          },
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#dddddd",
          },
          false,
          false,
          undefined,
          undefined,
        ]
      }
      underlayColor="#dddddd"
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#000000",
              },
              false,
              false,
            ]
          }
        >
          Modal.prompt (login-password)
        </Text>
      </View>
    </>
    <View
      style={
        Array [
          Object {
            "height": 9,
          },
          undefined,
        ]
      }
    />
    <
      activeOpacity={0.4}
      disabled={false}
      onHideUnderlay={[Function]}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      onShowUnderlay={[Function]}
      pressIn={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "borderWidth": 1,
            "justifyContent": "center",
          },
          Object {
            "height": 47,
            "paddingLeft": 15,
            "paddingRight": 15,
          },
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#dddddd",
          },
          false,
          false,
          undefined,
          undefined,
        ]
      }
      underlayColor="#dddddd"
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#000000",
              },
              false,
              false,
            ]
          }
        >
          Modal.prompt (secure-text)
        </Text>
      </View>
    </>
  </View>
  <View
    style={
      Object {
        "zIndex": 999,
      }
    }
  />
  <View
    style={
      Object {
        "zIndex": 999,
      }
    }
  />
  <View
    style={
      Object {
        "zIndex": 999,
      }
    }
  />
</ScrollView>
`;
